lib/utility/recipe_puppy_parser.rb

```diff
@@ -0,0 +1,91 @@
+module Utility
+  class RecipePuppyParser
+    ALL_RECIPES_URL = 'http://www.recipepuppy.com/api/?q=&p='.freeze
+    BASE_URL = 'http://www.recipepuppy.com/api/'.freeze
+
+# API query, not a database query
+    def query_all_recipes_for(*ingredients)
+      begin
+        ing_search = ingredients.join('').gsub(/\s+/, "")
+        recipes = []
+        100.times do |page|
+          response = HTTParty.get("#{BASE_URL}?i=#{ing_search}&p=#{page + 1}")
```

Comment:
what if line 12 - line 20 were a method which accepted an array and a url, then made the API call & added the data to the array? This allows us to call a descriptively named method 100 times